includes/Products/Sync/Background.php

```diff
@@ -342,7 +250,9 @@ private function process_item_delete( $prefixed_retailer_id ) {
 	private function send_item_updates( array $requests ): array {
 		$facebook_catalog_id = facebook_for_woocommerce()->get_integration()->get_product_catalog_id();
 		$response            = facebook_for_woocommerce()->get_api()->send_item_updates( $facebook_catalog_id, $requests );
-		$handles             = ( isset( $response->handles ) && is_array( $response->handles ) ) ? $response->handles : [];
+		$response_handles    = $response->handles;
+		// For some reason isset( $response->handles ) is false while isset( $response_handles ) is true
```

Comment:
This is a tricky behaviour but it is known. The `isset` is not a function - it is a language construct and it works on variables only. In your case, the `$response->handles` is not a variable. `handles` is a property of the `$this->response_data` defined in the base class `JSONResponse`. You access the `handles` via the magic function `__get` from the `JSONResponse`. As it was mentioned `isset` is not able to operate on anything else than a variable and here you have a return value of a function.

Complicated for sure :) Your code is good and you can remove the comment.